cmds/webboot/webboot.go

```diff
@@ -162,14 +210,24 @@ func main() {
 				log.Fatalf("Download option failed:%v", err)
 			}
 		case *ISO:
-			if err = entry.(*ISO).exec(ui.PollEvents(), *boot); err != nil {
+			if err = entry.(*ISO).exec(ui.PollEvents(), *dryRun); err != nil {
 				log.Fatalf("ISO option failed:%v", err)
 			}
 			entry = nil
 		case *DirOption:
-			if entry, err = entry.(*DirOption).exec(ui.PollEvents()); err != nil {
+			dirOption := entry.(*DirOption)
+			if entry, err = dirOption.exec(ui.PollEvents()); err != nil {
 				log.Fatalf("Directory option failed:%v", err)
 			}
+			if _, ok := entry.(*BackOption); ok {
+				// if dirOption.path == cacheDir means current dir is the root of cache dir
+				// and it should go back to the main menu.
+				if dirOption.path == cacheDir {
+					entry = getMainMenu(cacheDir)
```

Comment:
Hm so it seems like right now, every time the back option is chosen, getMainMenu will be called and display a menu that says "Weboot: choose an option" and contains the download option as well, right? But ideally 'back' just means go back to the menu page I just saw, not the main menu right?

Instead of having do to this, could you just set entry equal to that DirOption you're creating and leave it at that? Then you'd continue in the loop and display the same menu you would've displayed for the parent right? That takes out the need for creating this getMainMenu function and makes it so that users arent always seeing the main menu if they just want to go back.

Does that work?

Reply:
no it just be called when the current path is equal to the path of cache dir.  since at that time , if I still go back to the father folder, I will be out side of the cache dir, which is not what I want. And at that time , I should back to the main menu right? Otherwise when the current fir is not the root of cache dir it will go back to it's father folder

Reply:
May be I should record a video to show this process

Reply:
Got it, I see what I misunderstood - could you explain why you break after calling getMainMenu? If you break then how is the entry that comes back from getMainMenu handled?

Reply:
since getMainMenu(cacheDir) will returns the entry chooed in th main menu. and I user entry = getMainMenu(cacheDir) get this return. that is the same as what happened in other case: that is,  in every turn, I update my entry variable from the last turn's choose to the current tern's choose and  then begin the next turn.